galsim/chromatic.py

```diff
@@ -130,6 +135,210 @@ def __init__(self, obj):
+        required precision has been reached.  Also note that after calling `setupInterpolation`, it
+        is possible to revert to the exact calculation by calling `noInterpolation`.
+
+        The input parameter `waves` determines the input grid on which images are precomputed.  It
+        is difficult to give completely general guidance as to how many wavelengths to choose or how
+        they should be spaced; some experimentation compared with the exact calculation is warranted
+        for each particular application.  The best choice of settings might depend on how strongly
+        the parameters of the object depend on wavelength.
+
+        @param waves            The list, tuple, or NumPy array of wavelengths to be used when
+                                building up the grid of images for interpolation.  The wavelengths
+                                should be given in nanometers, and they should span the full range
+                                of wavelengths covered by any bandpass to be used for drawing Images
+                                (i.e., this class will not extrapolate beyond the given range of
+                                wavelengths).  They can be spaced any way the user likes, not
+                                necessarily linearly, though interpolation will be linear in
+                                wavelength.
+        @param oversample_fac   Factor by which to oversample the stored profiles compared to the
+                                default, which is to sample them at the Nyquist frequency for
+                                whichever wavelength has the highest Nyquist frequency.
+                                `oversample_fac`>1 results in higher accuracy but costlier
+                                pre-computations (more memory and time). [default: 1]
+        """
+        # Set up the interpolation (which can be a costly step, depending on the length of
+        # `waves`).
+        self.waves = np.sort(np.array(waves))
+
+        # Make the objects between which we are going to interpolate.  Note that these do not have
+        # to be saved for later, unlike the images.
+        objs = [ self.evaluateAtWavelength(wave) for wave in waves ]
+
+        # The evaluation step (above) led to the incorporation of any shears and other
+        # transformations into the stored models between which we interpolation, so reset the
+        # internal attributes that store information about transformations, if indeed there were any
+        # transformations.
+        transform_done = False
+        for wave in waves:
+            if hasattr(self, '_A') and not self._nullTransformation(wave):
+                transform_done = True
+                break
+        if transform_done and hasattr(self, '_A'):
+            self._A = lambda w: np.matrix(np.identity(3), dtype=float)
+            self._fluxFactor = lambda w: 1.0
+            self._reset = True
+
```

Comment:
Josh - I just made two commits to try to do this, but it's not working properly.  I played around with it for quite a while but am stumped.  (Note: there's a failing unit test that demonstrates exactly how it's going wrong.  The short answer is that I tried applying a chromatic transformation without any interpolation, then I did the case where I set up interpolation and applied a chromatic transformation, which is supposed to just unset the interpolation and do the normal way of drawing the transformed object.  These results should therefore be identical, but they don't seem to be.)

Is there something obvious that I'm missing here?  Sorry to dump this on you but I've been staring at the code pointlessly for about an hour since I initially wrote it, and am simply drawing a blank.